tests/test_engine.py

```diff
@@ -155,6 +159,9 @@ def item_error(self, item, response, spider, failure):
     def item_scraped(self, item, spider, response):
         self.itemresp.append((item, response))
 
+    def bytes_received(self, data, request):
+        self.bytes[request] += data
```

Comment:
I know this is only tests, but += is O(N^2) for bytes in Python 3. It is O(N) in CPython for str (because of some bytecode tricks), and it used to be O(N) in Python 2 for bytes, but this optimization was lost in transition.